Source/WTF/wtf/unicode/UTF8Conversion.cpp

```diff
@@ -127,6 +128,28 @@ bool convertUTF8ToUTF16ReplacingInvalidSequences(const char* source, const char*
     return convertUTF8ToUTF16Impl<true>(source, sourceEnd, targetStart, targetEnd, sourceAllASCII);
 }
 
+ComputeUTFLengthsResult computeUTFLengths(const char* sourceStart, const char* sourceEnd)
+{
+    size_t lengthUTF8 = sourceEnd - sourceStart;
+    size_t lengthUTF16 = 0;
+    char32_t orAllData = 0;
+    ConversionResult result = ConversionResult::Success;
+    size_t sourceOffset = 0;
+    while (sourceOffset < lengthUTF8) {
+        char32_t character;
+        size_t nextSourceOffset = sourceOffset;
+        U8_NEXT(reinterpret_cast<const uint8_t*>(sourceStart), nextSourceOffset, lengthUTF8, character);
```

Comment:
Should study `U8_NEXT` to be sure this needs the `reinterpret_cast`. Would be nice to get rid of it.

Reply:
The `reinterpret_cast` is not needed. I am removing it and the others in this file.